CLI: A panic is thrown when running multiple ui sessions using the same port

When trying to start a UI session using the same port as another active session, a panic is thrown.

STR:
1. In one terminal start a UI session using port `9999`
`./bin/darwin-amd64/connect-client account-ui --listen=localhost:9999`
2. In another terminal start a second UI session with the same port.

A panic will be thrown:
```
> ../bin/darwin-amd64/connect-client account-ui --listen=localhost:9999
time="2023-07-06T13:39:15.485Z" level=info msg="Client version" version=a344699
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x13f1020]

goroutine 1 [running]:
github.com/rstudio/connect-client/internal/services/api.(*Service).Run(0xc000113c20)
	/work/internal/services/api/http_service.go:109 +0xa0
github.com/rstudio/connect-client/cmd/connect-client/commands.(*accountUICmd).Run(0x2?, 0xc0001cb358?, 0x1?)
	/work/cmd/connect-client/commands/accounts.go:118 +0xc5
reflect.Value.call({0x144bee0?, 0xc0000d4c98?, 0x104b516?}, {0x14c6156, 0x4}, {0xc0001ecdb0, 0x2, 0x2?})
	/usr/local/go/src/reflect/value.go:586 +0xb0b
reflect.Value.Call({0x144bee0?, 0xc0000d4c98?, 0x145dbe0?}, {0xc0001ecdb0?, 0x14c5680?, 0xc00010dbe0?})
	/usr/local/go/src/reflect/value.go:370 +0xbc
github.com/alecthomas/kong.callMethod({0x14c5f83, 0x3}, {0x1467200?, 0xc0000d4c98?, 0x3?}, {0x144bee0?, 0xc0000d4c98?, 0xc00010dd98?}, 0x10158d0?)
	/work/vendor/github.com/alecthomas/kong/callbacks.go:95 +0x4fa
github.com/alecthomas/kong.(*Context).RunNode(0xc0000c6900, 0xc0001140f0, {0xc00010df48, 0x1, 0x1})
	/work/vendor/github.com/alecthomas/kong/context.go:755 +0x60f
github.com/alecthomas/kong.(*Context).Run(0x1467200?, {0xc00010df48?, 0xc0000acf80?, 0xc000092011?})
	/work/vendor/github.com/alecthomas/kong/context.go:780 +0x14e
main.main()
	/work/cmd/connect-client/main.go:81 +0x448
```

--- a/internal/services/api/http_service.go
+++ b/internal/services/api/http_service.go
@@ -106,6 +106,10 @@ func (svc *Service) Run() error {
 
 	// Open listener first so the browser can connect
 	listener, err := net.Listen("tcp", svc.listen)
+	if err != nil {
+		return fmt.Errorf("UI server error: %s", err)
+	}
+
 	svc.addr = listener.Addr()
 
 	// Log without the token

